src/nupic/algorithms/Connections.hpp

```diff
@@ -41,11 +41,11 @@ namespace nupic {
 namespace algorithms {
 namespace connections {
 
-typedef UInt32 CellIdx;
-typedef UInt16 SegmentIdx; /** Index of segment in cell. */
-typedef UInt16 SynapseIdx; /** Index of synapse in segment. */
-typedef UInt32 Segment;    /** Index of segment's data. */
-typedef UInt32 Synapse;    /** Index of synapse's data. */
+typedef UInt32 CellIdx;  //TODO instead of typedefs, use templates for proper type-checking
```

Comment:
typedefs have the disadvantage of only renaming the type for the user(=programmer), for proper type-checking (so that compiler checks difference for SegmentIdx, SynapseIdx) we'd need to use templates instead. 
I'll consider that in subsequent PR.